sensor/common/registry/registry_store.go

```diff
@@ -77,32 +68,37 @@ type CheckTLS func(ctx context.Context, origAddr string) (bool, error)
 // NewRegistryStore creates a new registry store.
 // The passed-in CheckTLS is used to check if a registry uses TLS.
 // If checkTLS is nil, tlscheck.CheckTLS is used by default.
-func NewRegistryStore(checkTLS CheckTLS) *Store {
-	regFactory := registries.NewFactory(registries.FactoryOptions{
+func NewRegistryStore(checkTLSFunc CheckTLS) *Store {
+	if checkTLSFunc == nil {
+		checkTLSFunc = tlscheck.CheckTLS
+	}
+	tlsCheckCache := newTLSCheckCache(checkTLSFunc)
+
+	defaultFactory := registries.NewFactory(registries.FactoryOptions{
 		CreatorFuncs: registries.AllCreatorFuncsWithoutRepoList,
 	})
 
+	factory := defaultFactory
+	if features.SensorLazyTLSChecks.Enabled() {
+		factory = newLazyFactory(tlsCheckCache)
```

Comment:
this only has the ability to make Docker and Red Hat registries. What about the rest?

Reply:
Those are the only two possible registry types we 'auto-create' Sensor.

 https://github.com/stackrox/stackrox/blob/906f9bdf0e57c8dfe54f3458daebc8bd85ade3e9/sensor/common/registry/registry_store.go#L156-L159

We could add others, to do so we'd need a way to 'detect' the type (easy with redhat registry since it has known endpoints), the others not as much.  And most the others types re-use the docker type so may no be much value in doing that.  When we get into a scenario where we need to customize the type because connectivity isn't working, perhaps we consider adding more then.

Our scanners don't have special logic for one registry type vs. another, don't believe k8s/runtime does either (could be wrong on that) - so perhaps we should phase these out of Sensor all together?

Reply:
can you add `Those are the only two possible registry types we 'auto-create' Sensor.` as a comment here so it's clear to future readers?